docs/src/fock.md

```diff
@@ -0,0 +1,147 @@
+
+- `vac = F₀ = F0` $=|0\rangle$ in the number state representation,
+- `F₁ = F1` $=|1\rangle$ in the number state representation.
+
+To create quantum analogues of a classical harmonic oscillator, or monochromatic electromagnetic waves, we can define a coherent state $|\alpha\rangle$, where $\alpha$ is a complex amplitude, with `CoherentState(α::Number, basis::Basis)`:
+
+```jldoctest
+julia> b = FockBasis(Inf, 0.0);
+
+julia> c = CoherentState(im, b)
+|im⟩
+```
+
+## Operators
+
+Operations on fock states are supported, and can be simplified with `qsimplify` and its rewriter `qsimplify_fock`. For instance, we can apply the raising (creation) $a^{\dagger}$ and lowering (annihilation or destroy) $a$ operators on a fock state as follows:
+
+```jldoctest
+julia> b = FockBasis(Inf, 0.0);
+
+julia> f = FockState(3, b);
+
+julia> raise = Create*f
+a†|3⟩
+
+julia> qsimplify(raise, rewriter=qsimplify_fock)
+2.0|4⟩
+
+julia> lower = Destroy*f
+a|3⟩
+
+julia> qsimplify(lower, rewriter=qsimplify_fock)
+1.7320508075688772|2⟩
+```
+Or, we can apply the number operator $\hat{n}$ to our fock state:
+
+```jldoctest
+julia> b = FockBasis(Inf, 0.0);
+
+julia> f = FockState(3, b);
+
+julia> num = N*f
+n|3⟩
+
+julia> qsimplify(num, rewriter=qsimplify_fock)
+3|3⟩
+```
+
+In the infinite dimension case for Fock states, constants are defined for number and ladder operators:
+
+- `N = n̂` $=\hat{n}$,
+- `Create = âꜛ` $=\hat{a}^{\dagger}$,
+- `Destroy = â` $=\hat{a}$.
+
+Phase-shift $U(\theta)$ and displacement $D(\alpha)$ operators, defined respectively as 
+$$U(\theta) = \exp\left(-i\theta\hat{n}\right) \quad \text{and} \quad D(\alpha) = \exp\left(\alpha\hat{a}^{\dagger} - \alpha\hat{a}\right),$$
+can be defined. Consider the following example:
+
+```jldoctest
+julia> b = FockBasis(Inf, 0.0);
+
+julia> displace = DisplaceOp(im, b)
+D(im)
+
+julia> c = qsimplify(displace*vac, rewriter=qsimplify_fock)
+|im⟩
+
+julia> phase = PhaseShiftOp(pi, b)
+U(π)
+
+julia> qsimplify(phase*c, rewriter=qsimplify_fock)
+|1.2246467991473532e-16 - 1.0im⟩
+```
+Here, we generated a coherent state $|i\rangle$ from the vacuum state $|0\rangle$ by applying the displacement operator defined by `DisplaceOp`. Then, we shifted its phase by $\pi$ with the phase shift operator (which is called with `PhaseShiftOp`) to get the result $|-i\rangle$.
+
+Summarized below are supported operators, which can be defined in any Fock basis.
+
+- Number operator: `NumberOp(basis:Basis)`,
+- Creation operator: `CreateOp(basis::Basis)`,
+- Annihilation operator: `DestroyOp(basis::Basis)`,
+- Phase-shift operator: `PhaseShiftOp(phase::Number, basis:Basis)`,
+- Displacement operator: `DisplaceOp(alpha::Number, basis::Basis)`.
+
+## Numerical Conversions to QuantumOptics.jl
+
+Fock systems can be translated to the ket representation with `express`. For instance:
+
+```jldoctest
+julia> using QuantumOptics
+
+julia> b = FockBasis(3);
```

Comment:
I think it makes more sense to not permit the creation of symbolic expressions with FockBasis with finite cutoff. These have only numeric meaning, symbolically a FockBasis has to be infinite. Rather the translation should have a configuration for how to do the cutoff.